api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -233,6 +233,16 @@ public Map<String, Integer> getAliases() {
     return aliasToId;
   }
 
+  /**
+   * Returns a map for this schema between field id and qualified field names. Initializes the map,
+   * if it has not been initialized by calls to {@link #findColumnName(int)}.
+   *
+   * @return a map of field id to qualified field names
+   */
+  public Map<Integer, String> idToName() {
```

Comment:
I'm generally a bit nervous about expanding the api here, is this better than just re-using findColumnName(id)? 

That said I don't think lazyIdToName() is that dangerous, but the java doc for this method contains a lot of very specific implementation details which should probably be removed.

Reply:
Yea I had a memory of discussing this with @aokolnychyi but cant remember now for sure :)

From looking now, I guess I need the list of ids as well, and so thought might as well return the map (as laziness wise its just as expensive to compute list of ids as the whole map).  I updated the comment, but let me know if you prefer something else.

Reply:
If Anton voted to expand the API then I'm onboard.